### PR TITLE
[PURCHASE-1992] Add logic for customizing message sent timestamps

### PR DESCRIPTION
**Pull request by jpotts244**
Addresses: https://artsyproduct.atlassian.net/browse/PURCHASE-1992

Did a lot of manual testing by changing the machine time on my computer locally

Ex of changes:

If last message sent less than 1 minute ago:
Timestamp reads `Just now`
Screen Shot 2020-07-06 at 10 28 12 AM

If last message sent less than 5 hours ago:
Timestamp reads `X hours ago`
Screen Shot 2020-07-06 at 6 34 40 PM

If last message sent less than 24 hours ago:
Timestamp reads `1:00PM` format
Screen Shot 2020-07-07 at 6 35 00 AM

If last message sent less than a week ago:
Timestamp reads `X days ago` format
Screen Shot 2020-07-09 at 6 35 21 AM

If last message sent less than a month ago:
Timestamp reads `1 week/X weeks ago` format
Screen Shot 2020-07-06 at 10 28 12 AM

If last message sent greater than a month ago:
Timestamp reads `MM/DD/YYYY` format
Screen Shot 2020-07-06 at 10 16 56 AM

### CODE REVIEW

File: src/v2/Apps/Conversation/Components/TimeSince.tsx [hunk 1]
```diff
@@ -35,9 +35,38 @@ const exactDate = (time: string) => {
...
+  return Math.floor(Math.abs(date.diffNow("minutes").toObject().minutes))
+}
+
 const relativeDate = (time: string) => {
   if (!time) return null
-  return DateTime.fromISO(time).toRelative()
+  const date = DateTime.fromISO(time)
+  const minutesSince = minutesSinceDate(date)
+
+  if (minutesSince <= 1) {
```
Comment (jpotts244) [inline src/v2/Apps/Conversation/Components/TimeSince.tsx:51]: Let me know what you think here - I tried to keep all handling of formatting contained close together within one function for explicitness.

`relativeDate` will always accept a timestamp and return a formatted string.

File: src/v2/Apps/Conversation/Components/TimeSince.tsx [hunk 2]
```diff
@@ -35,9 +35,38 @@ const exactDate = (time: string) => {
   }
 }
 
+const minutesSinceDate = (time: string | DateTime): number => {
+  if (!time) {
+    return null
+  }
+  const date = typeof time === "string" ? DateTime.fromISO(time) : time
+  return Math.floor(Math.abs(date.diffNow("minutes").toObject().minutes))
```
Comment (sepans) [inline src/v2/Apps/Conversation/Components/TimeSince.tsx:43]: #minor is that `toObject()` necessary? seems luxon `Duration` [has](https://moment.github.io/luxon/docs/class/src/duration.js~Duration.html#instance-get-minutes) minutes getter

Comment (jpotts244) [reply src/v2/Apps/Conversation/Components/TimeSince.tsx]: removed!

File: src/v2/Apps/Conversation/Components/__tests__/TimeSince.jest.tsx [hunk 3]
```diff
@@ -0,0 +1,51 @@
+import { relativeDate } from "../TimeSince"
+
+describe("relativeDate", () => {
+  beforeEach(() => {
+    const dateSpy = jest.spyOn(Date, "now")
+    dateSpy.mockReturnValue(1594072253087)
```
Comment (sepans) [inline src/v2/Apps/Conversation/Components/__tests__/TimeSince.jest.tsx:6]: 💯

**Comment by sepans** [comment]
This looks great! 🔥 

Thanks for adding the specs! Going to merge this

**Comment by xtina-starr** [comment]
Nice work!!

**Pull request merged**